java/server/src/org/openqa/selenium/netty/server/NettyServer.java

```diff
@@ -92,10 +109,18 @@ public void stop() {
 
   public NettyServer start() {
     ServerBootstrap b = new ServerBootstrap();
-    b.group(bossGroup, workerGroup)
-      .channel(NioServerSocketChannel.class)
-      .handler(new LoggingHandler(LogLevel.INFO))
-      .childHandler(new SeleniumHttpInitializer(handler));
+
+    if (secure) {
+      b.group(bossGroup, workerGroup)
+        .channel(NioServerSocketChannel.class)
+        .handler(new LoggingHandler(LogLevel.INFO))
+        .childHandler(new SeleniumHttpsInitializer(handler, sslCtx));  
```

Comment:
Why not just modify the existing `SeleniumHttpInitializer` to accept a `null` `sslCtx` and have it only add the SSL handler if the value is not null?